snowplow-tracker/src/main/java/com/snowplowanalytics/core/utils/Util.kt

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (c) 2015-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.core.utils
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.Location
+import android.location.LocationManager
+import android.net.ConnectivityManager
+import android.os.Build
+import android.util.Base64
+import com.snowplowanalytics.core.constants.Parameters
+import com.snowplowanalytics.core.constants.TrackerConstants
+import com.snowplowanalytics.core.tracker.Logger
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+import java.io.*
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * Provides basic Utilities for the Snowplow Tracker.
+ */
+object Util {
+    private val TAG = Util::class.java.simpleName
+
+    /**
+     * Returns the current System time as a String.
+     *
+     * @return the system time as a string
+     */
+    @JvmStatic
+    val timestamp: String
+        get() = System.currentTimeMillis().toString()
+
+    @JvmStatic
+    fun getDateTimeFromTimestamp(timestamp: Long): String {
+        val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale("en"))
+        dateFormat.timeZone = TimeZone.getTimeZone("UTC")
+        val date = Date(timestamp)
+        return dateFormat.format(date)
+    }
+
+    /**
+     * Encodes a string into Base64.
+     *
+     * @param string the string too encode
+     * @return a Base64 encoded string
+     */
+    @JvmStatic
+    fun base64Encode(string: String): String {
+        return Base64.encodeToString(string.toByteArray(), Base64.NO_WRAP)
+    }
+
+    /**
+     * Generates a random UUID for each event.
+     *
+     * @return a UUID string
+     */
+    @JvmStatic
+    val uUIDString: String
```

Comment:
Nit: For things that generate a different value on each access, I prefer to use functions instead of properties. With properties, there is an expectation that the value is being held and remains consistent.